transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionSubsystemAdd.java

```diff
@@ -508,8 +509,16 @@ private void performCoordinatorEnvBoottime(OperationContext context, ModelNode c
         final boolean coordinatorEnableStatistics = TransactionSubsystemRootResourceDefinition.STATISTICS_ENABLED.resolveModelAttribute(context, coordEnvModel).asBoolean();
         final boolean transactionStatusManagerEnable = TransactionSubsystemRootResourceDefinition.ENABLE_TSM_STATUS.resolveModelAttribute(context, coordEnvModel).asBoolean();
         final int coordinatorDefaultTimeout = TransactionSubsystemRootResourceDefinition.DEFAULT_TIMEOUT.resolveModelAttribute(context, coordEnvModel).asInt();
+        final int maximumTimeout = TransactionSubsystemRootResourceDefinition.MAXIMUM_TIMEOUT.resolveModelAttribute(context, coordEnvModel).asInt();
 
-        ContextTransactionManager.setGlobalDefaultTransactionTimeout(coordinatorDefaultTimeout);
+        // WFLY-9955 Allow the timeout set to "0" while translating into the maximum timeout
+        TransactionSubsystemRootResourceDefinition.maximum_timeout = maximumTimeout;
```

Comment:
There are a few problems with this data sharing via a global static variable:
1) It is not thread safe, access needs to be synchronized or it needs to be final for it to share
2) In a domain where we can have several profiles, the last write will 'win'